src/Context/GameContext.js

```diff
@@ -0,0 +1,147 @@
+import React, { createContext, useState, useEffect } from 'react';
+import { gameData } from '../game-data';
+ 
+const GameContext = createContext();
+
+const GameProvider = ({ children }) => {
+  const [board, setBoard] = useState(gameData);
+  const [player, setPlayer] = useState('X');
+  const [message, setMessage] = useState('Time to play');
+  const [winner, setWinner] = useState('');
+  const [gameOver, setGameOver] = useState(false);
+  const [tie, setTie] = useState(false);
+
+  useEffect(() => {
+    const checkWinner = () => {
+      if (tie) {
+        setMessage('It is a tie!');
+      } else {
+        winner && gameOver ? setMessage(`${winner} wins this time!`) : setMessage(`${player} it's your turn`);
+      }
+    };
+    checkWinner();
+  }, [player, winner, gameOver, tie]);
+
+
+  function resetGameButton() {
+    setBoard(gameData);
```

Comment:
I would just plug in the game data here because you are changing the game data every with every click. This would allow for a board reset.